core/src/mender-artifact.c

```diff
@@ -392,7 +394,15 @@ mender_artifact_parse_tar_header(mender_artifact_ctx_t *ctx) {
     ctx->file.name = tmp;
 
     /* Retrieve file size */
-    sscanf(tar_header->size, "%o", (unsigned int *)&(ctx->file.size));
+    char *end_ptr;
+    errno          = 0; /* to distinguish between success/failure */
+    ctx->file.size = strtoul(tar_header->size, &end_ptr, 8);
```

Comment:
Here we are assuming `sizeof(size_t) >= sizeof(unsigned long)`. I guess that's a safe assumption, but an `assert()` would not hurt.

Reply:
The same applies below.